### ISSUE TITLE
Loss Functions

### ISSUE THREAD

**Issue by davidmascharka** [davidmascharka/MyNN#4]
A loss function should take model outputs and target values and return the loss. Something like:

``` python
# assuming model outputs in `outputs` and targets in `targets
loss_func = losses.L1()
loss = loss_func(outputs, targets)
```

### The Losses
- [x] L1
- [x] MSE
- [x] Negative log-likelihood
- [x] KL divergence
- [x] Cross-entropy
- [x] Balanced cross-entropy (weighting factor alpha on each class)
- [x] [Focal loss](https://arxiv.org/abs/1708.02002) [softmax and not]
- [x] Smooth L1 (Huber)
